emcc

```diff
@@ -1077,12 +1073,13 @@ try:
     return args
 
   # -E preprocessor-only support
-  if '-E' in newargs:
+  if '-E' in newargs or '-M' in newargs or '-MM' in newargs:
     input_files = map(lambda x: x[1], input_files)
     cmd = get_bitcode_args(input_files)
     if specified_target:
       cmd += ['-o', specified_target]
-    logging.debug('just preprocessor ' + ' '.join(cmd))
+    # Do not compile, but just output the result from preprocessing stage or output the dependency rule. Warning: clang and gcc behave differently with -MF! (clang seems to not recognize it)
+    logging.debug('just preprocessor ' if '-E' in newargs else 'just dependencies: ' + ' '.join(cmd))
```

Comment:
i like the use if `if-else`, but please add parentheses to clarify the order of operations.